src/Routing/RouteCollection.php

```diff
@@ -77,6 +78,12 @@ public function add(Route $route, array $options = [])
 
         // Explicit names
         if (isset($options['_name'])) {
+            if (isset($this->_named[$options['_name']])) {
+                throw new DuplicateNamedRouteException([
+                    'url' => $options['_name'],
+                    'message' => 'A named route was found for "%s" that is already used, route names must be unique across your entire application.'
```

Comment:
Do you need the message in both places? If we don't need both copies of the message, can we remove one?